### PR TITLE
Adding Permissions to User Struct

### PR DESCRIPTION
**Pull request by jeevanragula**
## Description

Get Account API doesn't return Permission details. So I am adding Permissions in the User struct.
This will fix https://github.com/hashicorp/go-tfe/issues/673

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" go test ./... -v -run TestFunctionsAffectedByChange

?       github.com/hashicorp/go-tfe/examples/configuration_versions     [no test files]
?       github.com/hashicorp/go-tfe/examples/organizations      [no test files]
?       github.com/hashicorp/go-tfe/examples/registry_modules   [no test files]
?       github.com/hashicorp/go-tfe/examples/users      [no test files]
?       github.com/hashicorp/go-tfe/examples/workspaces [no test files]
?       github.com/hashicorp/go-tfe/mocks       [no test files]
=== RUN   TestUsersReadCurrent
=== RUN   TestUsersReadCurrent/two_factor_options_are_decoded
=== RUN   TestUsersReadCurrent/permissions_are_decoded
--- PASS: TestUsersReadCurrent (1.67s)
    --- PASS: TestUsersReadCurrent/two_factor_options_are_decoded (0.00s)
    --- PASS: TestUsersReadCurrent/permissions_are_decoded (0.00s)
PASS
ok      github.com/hashicorp/go-tfe     1.854s


...
```

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ type Users interface {
 	// ReadCurrent reads the details of the currently authenticated user.
 	ReadCurrent(ctx context.Context) (*User, error)
 
-	// Update attributes of the currently authenticated user.
+	// UpdateCurrent Update attributes of the currently authenticated user.
```
Comment (brandonc) [inline user.go:21]:
```suggestion
	// UpdateCurrent updates attributes of the currently authenticated user.
```

File: user.go [hunk 2]
```diff
@@ -29,19 +29,32 @@ type users struct {
...
+	ID               string           `jsonapi:"primary,users"`
+	AvatarURL        string           `jsonapi:"attr,avatar-url"`
+	Email            string           `jsonapi:"attr,email"`
+	IsServiceAccount bool             `jsonapi:"attr,is-service-account"`
+	TwoFactor        *TwoFactor       `jsonapi:"attr,two-factor"`
+	UnconfirmedEmail string           `jsonapi:"attr,unconfirmed-email"`
+	Username         string           `jsonapi:"attr,username"`
+	V2Only           bool             `jsonapi:"attr,v2-only"`
+	IsSiteAdmin      bool             `jsonapi:"attr,is-site-admin"`
+	IsSsoLogin       bool             `jsonapi:"attr,is-sso-login"`
```
Comment (brandonc) [inline user.go:41]: These two attributes aren't always present-They are only returned for the current user (this struct is referenced by many other entities). For that reason, I think this should be a `*bool` type in order to allow nil values.

```suggestion
	IsSiteAdmin      *bool            `jsonapi:"attr,is-site-admin"`
	IsSsoLogin       *bool            `jsonapi:"attr,is-sso-login"`
```

**Review by brandonc — COMMENTED** [review]
Thank you for the very thorough contribution! My feedback is very minor and I'll be happy to merge it once it's addressed

**Comment by jeevanragula** [comment]
@brandonc Thanks for the suggestions. Updated the PR now.

**Review by brandonc — APPROVED** [review]

**Pull request merged**